README.md

```diff
@@ -0,0 +1,257 @@
+
+## Contributing
+
+Please read [CONTRIBUTING.md](CONTRIBUTING.md) for details on our code of conduct, and the process for submitting pull requests.
+
+## License
+
+- Ruby
+- PostgreSQL with TimescaleDB extension
+- Environment variables for API access and configuration:
+  - `AESO_TOKEN` - Alberta Electric System Operator API token
+  - `AESO_QUEUE_URL` - AWS SQS queue URL for AESO data processing
+  - `EIA_TOKEN` - U.S. Energy Information Administration API token
+  - `ELEXON_TOKEN` - GB Electricity Market API token
+  - `ENTSOE_TOKEN` - ENTSOE API token
+  - `ENTSOE_USER` - ENTSOE SFTP username
+  - `ENTSOE_PASSWORD` - ENTSOE SFTP password
+  - `ERCOT_PROXY_API_KEY` - ERCOT proxy API key
+  - `ES_URL` - Elasticsearch URL for logging (optional)
+  - `RAILS_ENV` - Environment name (development/test/production)
+  - `TAIPOWER_QUEUE_URL` - AWS SQS queue URL for Taipower data processing
+
+## Database Configuration
+
+Database configuration should be specified in `db/config.yml`. The system uses ActiveRecord with TimescaleDB for time-series data storage.
+
+## Testing
+
+The project uses RSpec for testing. Tests can be run with:
+
+```bash
+bundle exec rspec
+```
+
+Test coverage is tracked using SimpleCov with Cobertura formatter.
+
+## Technical Stack
+
+- Ruby with YJIT enabled
+- PostgreSQL with TimescaleDB extension for time-series data
+- Semantic Logger for structured logging
+- AWS SQS for queue processing
+- Elasticsearch for logging (optional)
+- Various HTTP clients (Faraday, HTTParty) for API access
+- Fast parsers (FastJsonparser, FastestCSV) for efficient data processing
+
+## Data Processing Components
+
+The system includes several key components:
+
+- Data collectors for each market operator
+- Time zone handling for different regions using TZInfo
+- Data validation and normalization
+- Bulk data import capabilities
+- Historical data processing
+- Real-time data updates
+- SFTP support for certain providers (e.g., ENTSOE)
+- AWS SQS integration for queue-based processing
+- Automatic data deduplication
+
+## Output Formats
+
+Data can be processed into various formats through the `Out2` module:
+
+- Generation data by fuel type
+- Unit capacity and details
+- Transmission flows
+- Price data
+- Load/demand data
+- Unit-level generation data
+- Rooftop solar data (where available)
+
+## Development
+
+When contributing:
+
+1. Write tests for new functionality (RSpec with VCR for HTTP mocking)
+2. Follow existing code patterns for new market integrations
+3. Use the provided CLI mixins for consistent command-line interfaces
+4. Handle time zones appropriately using the TZ constants
+5. Use the provided logging infrastructure (SemanticLogger)
+6. Leverage the fast parsers (FastJsonparser, FastestCSV) for data processing
+7. Follow the established data validation patterns
+
+## Rake Tasks
+
+The system provides various Rake tasks for data collection:
+
+```bash
+# ENTSOE data collection
```

Comment:
`rake -j4 all` runs all the data collection tasks which is what I use from Jenkinsfile and GH actions. Also runs multiple tasks in parallel via `-j4`.